src/sources/logplex.rs

```diff
@@ -0,0 +1,190 @@
+use crate::{
+    event::{self, Event},
+    topology::config::{DataType, GlobalOptions, SourceConfig},
+};
+use bytes::Buf;
+use chrono::{DateTime, Utc};
+use futures::{sync::mpsc, Future, Sink};
+use serde::{Deserialize, Serialize};
+use std::sync::{Arc, Mutex};
+use std::{
+    io::{BufRead, BufReader},
+    net::SocketAddr,
+};
+use stream_cancel::Tripwire;
+use warp::Filter;
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+pub struct LogplexConfig {
+    address: SocketAddr,
+}
+
+#[typetag::serde(name = "logplex")]
+impl SourceConfig for LogplexConfig {
+    fn build(
+        &self,
+        _: &str,
+        _: &GlobalOptions,
+        out: mpsc::Sender<Event>,
+    ) -> crate::Result<super::Source> {
+        let (trigger, tripwire) = Tripwire::new();
+        let trigger = Arc::new(Mutex::new(Some(trigger)));
+
+        let svc = warp::post2()
+            .and(warp::path("events"))
+            .and(warp::header::<usize>("Logplex-Msg-Count"))
+            .and(warp::header::<String>("Logplex-Frame-Id"))
+            .and(warp::header::<String>("Logplex-Drain-Token"))
+            .and(warp::body::concat())
+            .and_then(move |msg_count, frame_id, drain_token, body| {
+                info!(message = "Handling logplex request", %msg_count, %frame_id, %drain_token);
+
+                let events = body_to_events(body);
+
+                if events.len() != msg_count {
+                    error!(message = "Parsed event count does not match message count header", event_count = events.len(), %msg_count);
+                    debug_assert!(false, "Parsed event count does not match message count header");
```

Comment:
Can you explain why we need this?

Reply:
If we're in this branch, that means we've failed to parse the body correctly. In development, I want that to blow up, fail tests, etc. In production, I want to spit out an error but keep going so that we remain available for users.

Reply:
Would 

```rust
if cfg_is!(test) {
	panic!(...)
}
```
be more clear here than debug_assert? Mostly, so that people know this should panic in tests.